libs/dal/src/repositories/notification/notification.schema.ts

```diff
@@ -44,10 +44,13 @@ const notificationSchema = new Schema<NotificationDBModel>(
     payload: {
       type: Schema.Types.Mixed,
     },
+    expireAt: Schema.Types.Date,
   },
   schemaOptions
 );
 
+notificationSchema.index({ expireAt: 1 }, { expires: '48h' });
```

Comment:
@ainouzgali speaking with @Cliftonz it seems we only need to add the "Expires" path for self hosted right? the online archive with take care for it for us?

Reply:
I suspect it will not do any harm not to add the check. As the ttl deletion will only run 48 hours after the online archive.  If its already deleted, then ttl won't run. 
But I would like to make sure when we have the integration with the online archive, and I wouldn't mind adding it @Cliftonz @scopsy

Reply:
@ainouzgali I do not think we should have both. Both act as a ttl and if the backlog of the archival process gets two log then we could lose data.
I think we should do something like the following:
```
if Novu_Managed && Enable_TTL 
    enable ttl index
else
    only create index on expireAt for online archive

```
This way users who want to keep everything can and users who dont can have them deleted after the standard amount of time.
We can make community issues to add configurations for the ttl for each entity we are supporting.

Reply:
So we have two things here:
1. managed should not have 'expire'(ttl) - only by archive 
2. self hosted should have enable_ttl field -> meaning they will have the option to either have our settings for the ttl or to not have it at all 

Have I got that right?

Reply:
So basically for Managed Service:
```
messageSchema.index({ expireAt: 1 });
```

And for Self hosted:
```
messageSchema.index({ expireAt: 1 }, { expires: '48h' });
```